### PR TITLE
GRF-234: value_collection in the Category model

### PR DESCRIPTION
**Pull request by BryanS29-akeneo**
**Description (for Contributor and Core Developer)**

Retrieve value_collection from category table and format it in ValueCollection.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private function __construct(private ?array $values)
+    {
+    }
+
+    public static function fromArray(array $values): self
+    {
+        return new self($values);
+    }
+
+    public function getAttributeTextData(string $attributeCode, string $attributeIdentifier, string $localeCode): ?array
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php:22]: Maybe we can add comment to specify the return array structure

File: src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        return $this->values[sprintf('%s_%s_%s', $attributeCode, $attributeIdentifier, $localeCode)];
+    }
+
+    public function getAttributeData(string $attributeCode, string $attributeIdentifier): ?array
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php:27]: Maybe we can add comment to specify the return array structure

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php [hunk 3]
```diff
@@ -21,7 +22,7 @@ public function __construct(private Connection $connection)
     {
     }
 
-    public function byId(int $categoryId): ?Category
+    public function byId(int $categoryId, ?int $userId = null): ?Category
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php:25]: Why to put userId as null default ?
will be the case at 80% ?

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php [hunk 4]
```diff
@@ -30,7 +31,7 @@ public function byId(int $categoryId): ?Category
         return $this->execute($condition);
     }
 
-    public function byCode(string $categoryCode): ?Category
+    public function byCode(string $categoryCode, ?int $userId = null): ?Category
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php:34]: same as well about the default value.

File: src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/Sql/GetCategorySqlIntegration.php [hunk 5]
```diff
@@ -5,16 +5,51 @@
...
+use Akeneo\Category\Infrastructure\Storage\Sql\GetCategorySql;
 use Akeneo\Test\Integration\Configuration;
 use Akeneo\Test\Integration\TestCase;
+use Doctrine\DBAL\Connection;
 
 /**
  * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
 class GetCategorySqlIntegration extends TestCase
```
Comment (antoinetrouve) [inline src/Akeneo/Category/back/tests/Integration/Infrastructure/Storage/Sql/GetCategorySqlIntegration.php:17]: It could be good to add some test for the null return of the query ? I don't know if its necessary.

**Review by antoinetrouve — APPROVED** [review]

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php (hunk 3 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php:25]: I said that beacause of this : https://github.com/akeneo/pim-community-dev/pull/17263#discussion_r905403951. Apparently it is a bad practice to put default value.

File: src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php (hunk 4 above)
Comment (antoinetrouve) [reply src/Akeneo/Category/back/Infrastructure/Storage/Sql/GetCategorySql.php:34]: I said that because of this : https://github.com/akeneo/pim-community-dev/pull/17263#discussion_r905403951. Apparently it is a bad practice to put default value.

**Review by etiennekerbastard — APPROVED** [review]

File: src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php (hunk 2 above)
Comment (BryanS29-akeneo) [reply src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php:27]: It's a temporary function, it was to validate the integration test. It will be removed when we will have our attribute model.

File: src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php (hunk 1 above)
Comment (BryanS29-akeneo) [reply src/Akeneo/Category/back/Domain/ValueObject/ValueCollection.php:22]: It's a temporary function, it was to validate the integration test. It will be removed when we will have our attribute model.
